packages/core/src/components/forms/_controls.scss

```diff
@@ -367,8 +373,8 @@ $control-indicator-spacing: $pt-grid-size !default;
     }
 
     input:checked ~ .#{$ns}-control-indicator::before {
-      // 1em is size of indicator
-      left: $switch-width - 1em;
+      right: 0;
+      left: auto;
```

Comment:
this breaks the animation 😢 because it switches properties from `left` to `right`

try this: `left: calc(100% - 1em);` 🏆 
(https://github.com/palantir/blueprint/pull/2721 for context on why that works)